src/xtgeo/grid3d/grid.py

```diff
@@ -141,100 +257,173 @@ class Grid(_Grid3D):
+        units: Optional[Units] = None,
+        filesrc: Optional[Union[Path, str]] = None,
+        props: Optional[GridProperties] = None,
+        name: Optional[str] = None,
+        roxgrid=None,
+        roxindexer=None,
+    ):
+        """This function only serves to allow deprecated initialization."""
+        # TODO: Remove once implicit initialization such as Grid().from_file()
+        # is removed
+        self._xtgformat = xtgformat
+        if xtgformat == 1:
+            if ncol is None or nrow is None or nlay is None:
+                raise ValueError("Need to give dimensions when xtgformat=1")
+            self._ncol = ncol
+            self._nrow = nrow
+            self._nlay = nlay
+        else:
+            self._ncol = actnumsv.shape[0]
+            self._nrow = actnumsv.shape[1]
+            self._nlay = actnumsv.shape[2]
 
-            geo = xtgeo.Grid()
-            geo.from_file("myfile.roff")
+        self._coordsv = coordsv
+        self._zcornsv = zcornsv
+        self._actnumsv = actnumsv
+        self._dualporo = dualporo
+        self._dualperm = dualperm
 
-            # alternative (make instance directly from file):
-            geo = xtgeo.Grid("myfile.roff")
+        self._filesrc = filesrc
 
-            # or use
-            geo = xtgeo.grid_from_file("myfile.roff")
-        """
-        super().__init__()
-
-        self._coordsv = None  # numpy array to coords vector
-        self._zcornsv = None  # numpy array to zcorns vector
-        self._actnumsv = None  # numpy array to actnum vector
-
-        # _xtgformat: internal flag for storage structure. 1 is the "current" while 2
-        # will be the new one. This will affect how _coordsv _zcornsv and _actnumsv
-        # are organized! The corresponding C routines for 1: grd3d_*, while 2: grdcp3d_*
-        self._xtgformat = 2
-
-        # the following block is currently not in use, but required for metadata
-        self._xshift = 0.0
-        self._yshift = 0.0
-        self._zshift = 0.0
-        self._xscale = 1.0
-        self._yscale = 1.0
-        self._zscale = 1.0
-
-        self._filesrc = None
-
-        self._props = None  # None or a GridProperties instance
-        self._name = "noname"
-        self._subgrids = None  # A python dict if subgrids are given
+        if props is None:
+            self._props = GridProperties(ncol=self.ncol, nrow=self.nrow, nlay=self.nlay)
+        else:
+            self._props = props
+        self._name = name
+        self._subgrids = subgrids
         self._ijk_handedness = None
 
-        # Simulators like Eclipse may have a dual poro/perm model
-        self._dualporo = False
-        self._dualperm = False
-        self._dualactnum = None  # will be a GridProperty()
+        self._dualactnum = None
+        if dualporo:
+            self._dualactnum = self.get_actnum(name="DUALACTNUM")
+            acttmp = self._dualactnum.copy()
+            acttmp.values[acttmp.values >= 1] = 1
```

Comment:
I can see that this is the old behaviour as well, I'm just wondering if this warrants some warning?